### PR TITLE
PLANET 3411 enform shortcode

### PR DESCRIPTION
**Pull request by angtheod**
Create a shortcode for EN Form Posts which will render a form with the Editor selected components (fields, questions, optins).

example: `[p4en_form id="<post_id>" /]`

### CODE REVIEW

**Comment by angtheod** [comment]
The new Block is still not rendering properly in the backend, but this shouldn't block us. We could fix that later. The EN form is rendering though inside its own EN Form post.

File: classes/controller/blocks/class-enblock-controller.php [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+						}
+						$options[] = [
+							'label'   => Pages_Datatable_Controller::SUBTYPES[ $type ]['subType'],
+							'options' => $group_options,
+						];
+					}
+				}
+
+				$query = new \WP_Query( [
+					'post_status' => 'publish',
```
Comment (sagarsdeshmukh) [inline classes/controller/blocks/class-enblock-controller.php]: Array double arrow alignment.(PHPCS)

File: classes/controller/blocks/class-enblock-controller.php [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+			$data = [];
+
+			if ( isset( $fields['thankyou_url'] ) && 0 !== strpos( $fields['thankyou_url'], 'http' ) ) {
+				$fields['thankyou_url'] = 'http://' . $fields['thankyou_url'];
+			}
+
+			$data = array_merge(
+				$data,
+				[
+					'fields'          => $fields,
```
Comment (sagarsdeshmukh) [inline classes/controller/blocks/class-enblock-controller.php]: extra spaces found in array alignment.

File: classes/controller/menu/class-enform-post-controller.php [hunk 3]
```diff
@@ -146,16 +147,73 @@ public function modify_post_row_actions( $actions, $post ) {
...
+		/**
+		 * Adds shortcode for this custom post type.
+		 *
+		 * @param array $atts Array of attributes for the shortcode.
+		 */
+		public function handle_form_shortcode( $atts ) {
+			global $pagenow;
+
+			// Define attributes and their defaults.
+			$atts = shortcode_atts( [
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: `PHPCS : Opening parenthesis of a multi-line function call must be the last content on the line`

File: classes/controller/menu/class-enform-post-controller.php [hunk 4]
```diff
@@ -186,6 +244,36 @@ public function view_meta_box_selected( $post ) {
...
+		 */
+		public function view_fields_meta_box( $post ) {
+			$list_table = new Enform_Fields_List_Table();
+			$list_table->prepare_items();
+			$list_table->display();
+		}
+
+		/**
+		 * Adds a meta box for the EN questions.
+     *
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-enform-post-controller.php]: Function doc block line misaligned here.

**Comment by sagarsdeshmukh** [comment]
@angtheod - I added a few comments. remaining functionality looks good to me 👍

**Pull request merged**